### ISSUE TITLE
Concerning the map editor

### ISSUE THREAD

**Issue by ShadoW2551** [VladimirMakeev/D2ModdingToolset#239]
Hi
Is it possible to add another category in the map editor under "terrain" - (special terrains), for other objects like "other" or something similar?
Is there a way to add sound to our object?

**Comment by VladimirMakeev** [comment]
> Is it possible to add another category in the map editor under "terrain" - (special terrains), for other objects like "other" or something similar?

I already explained new terrains in this [issue](https://github.com/VladimirMakeev/D2ModdingToolset/issues/238#issuecomment-1893428199).

> Is there a way to add sound to our object?

Currently no, but I will support sounds for a new strategic map object when I finish implementing it.

Please, don't edit the issues, create a new one instead. Or at least describe what happened to `battle.carryXpOverUpgrade` setting - should I spent time figuring out if needs fixing or not.

**Comment by ShadoW2551** [comment]
> > Is it possible to add another category in the map editor under "terrain" - (special terrains), for other objects like "other" or something similar?
> 
> I already explained new terrains in this [issue](https://github.com/VladimirMakeev/D2ModdingToolset/issues/238#issuecomment-1893428199).
> 
> > Is there a way to add sound to our object?
> 
> Currently no, but I will support sounds for a new strategic map object when I finish implementing it.
> 
> Please, don't edit the issues, create a new one instead. Or at least describe what happened to `battle.carryXpOverUpgrade` setting - should I spent time figuring out if needs fixing or not.

Going back to "battle.carryXpOverUpgrade" it works well on "no lv restrictions on heroes" maps. On the other hand, "with restrictions" our hero will reach 99% PD "in the background", sometimes you will see how much he has gained in battles. When we start another map with the exported hero, he will start with these 99% PD. 
As for the "Main" question, what I meant was whether it is possible to add more categories than the 4 default to our map editor. 
Last question: is there a limit to the objects that can be added to the game, for example, 10 thousand?

**Comment by VladimirMakeev** [comment]
>  it is possible to add more categories than the 4 default to our map editor

You mean add another button along with 'Quest Rules', 'Map', 'Objects' and 'Events'? Yes, it's possible if we support it in the modding toolset, but no one asked for this.

> is there a limit to the objects that can be added to the game

There is a hard limit of 65536 unique objects of the same type in the game: units or items in scenario, for example.
There is also another hard limit inside rendering system - how many objects on screen can be shown (more correctly speaking: how many render commands can fit into queue).
There is also another hard limit because of internal network message buffer size. By default it is 500Kb, we made it bigger but the restriction is still here. If by some circumstances game decides to send an 'update objects' message that is bigger than internal buffer - things will break.
There also was a limit of max 200 stacks on the map in Scenario Editor but we disabled it in modding toolset.
Mapmakers know empirically how much content they could place without issues, they also test their maps before releasing to the public.
